Heimdall/Heimdall.swift

```diff
@@ -95,6 +95,19 @@ public class Heimdall {
             completion(result.map { _ in return })
         }
     }
+    
+    /// Requests an access token with the given grant type URI and parameters
+    ///
+    /// **Note:** The completion closure may be invoked on any thread.
+    ///
+    /// :param: grantType The grant type URI of the extension grant
+    /// :param: parameters The required parameters for the external grant
+    /// :param: completion A callback to invoke when the request completed.
+    public func requestAccessToken(#grantType: String, parameters: [String: String], completion: Result<Void, NSError> -> ()) {
```

Comment:
I wonder whether we should expose the generic `requestAccessToken` method for future extensions.. opinions?